src/components/FilterAttachedResults/FilterAttachedResults.ts

```diff
@@ -0,0 +1,149 @@
+import {
+    Component,
+    ComponentOptions,
+    IComponentBindings,
+    $$,
+    Initialization,
+    Checkbox,
+    QueryEvents,
+    IBuildingQueryEventArgs,
+    QueryStateModel,
+    load,
+    IAttributesChangedEventArg
+} from 'coveo-search-ui';
+
+export interface IAnalyticsAttachedResultsFilterMeta {
+    filteredAttachedResults: boolean;
+}
+
+export interface IFilterAttachedResultsOptions {
+    /** Specifies the text displayed next to the checkbox. */
+    text?: string;
+    /** The function that is called when retrieving uri hashes to filter. */
+    getUriHashes?: () => string[];
```

Comment:
Are we absolutely sure that we will always use the urihashes?
Where has permanentId gone?
Should we let people define their own fields?

Reply:
good point, we should let the user decide.. we could add an option to set the field and simply use that when building the query!

Reply:
I would say, can we have an option to decide which field to be used to uniquely identify results. But default to urihash?